src/workerd/server/workerd-api.c++

```diff
@@ -295,39 +295,33 @@ kj::Maybe<jsg::Bundle::Reader> fetchPyodideBundle(
     return kj::none;
   }
 
-  {
-    kj::Thread([&]() {
-      kj::String url =
-          kj::str("https://pyodide-capnp-bin.edgeworker.net/pyodide_", version, ".capnp.bin");
-      KJ_LOG(INFO, "Loading Pyodide bundle from internet", url);
-      kj::AsyncIoContext io = kj::setupAsyncIo();
-      kj::HttpHeaderTable table;
+  kj::String url =
+      kj::str("https://pyodide-capnp-bin.edgeworker.net/pyodide_", version, ".capnp.bin");
+  KJ_LOG(INFO, "Loading Pyodide bundle from internet", url);
+  kj::HttpHeaderTable table;
 
-      kj::TlsContext::Options options;
-      options.useSystemTrustStore = true;
+  kj::TlsContext::Options options;
+  options.useSystemTrustStore = true;
 
-      kj::Own<kj::TlsContext> tls = kj::heap<kj::TlsContext>(kj::mv(options));
-      auto& network = io.provider->getNetwork();
-      auto tlsNetwork = tls->wrapNetwork(network);
-      auto& timer = io.provider->getTimer();
+  kj::Own<kj::TlsContext> tls = kj::heap<kj::TlsContext>(kj::mv(options));
+  auto& network = io.provider->getNetwork();
+  auto tlsNetwork = tls->wrapNetwork(network);
+  auto& timer = io.provider->getTimer();
 
-      auto client = kj::newHttpClient(timer, table, network, *tlsNetwork);
+  auto client = kj::newHttpClient(timer, table, network, *tlsNetwork);
 
-      kj::HttpHeaders headers(table);
+  kj::HttpHeaders headers(table);
 
-      auto req = client->request(kj::HttpMethod::GET, url.asPtr(), headers);
+  auto req = client->request(kj::HttpMethod::GET, url.asPtr(), headers);
 
-      auto res = req.response.wait(io.waitScope);
-      KJ_ASSERT(res.statusCode == 200,
-          kj::str(
-              "Request for Pyodide bundle at ", url, " failed with HTTP status ", res.statusCode));
-      auto body = res.body->readAllBytes().wait(io.waitScope);
+  auto res = req.response.wait(io.waitScope);
```

Comment:
Once you convert this function to a promise you won't need the waitScope, you could just await.

Reply:
Oh wow, just looked closer at this. This whole hack needs to be refactored -- we shouldn't be blocking on I/O in a synchronous function, either using a thread or using `.wait()`.

I guess it happens to work today only because workerd constructs all `WorkerdApi` objects at startup anyway, so this only blocks startup. But that won't be true forever. I'm working on an API right now for dynamically loading isolates, which will need to construct new `WorkerdApi` instances at runtime.

I think that Python dependencies need to be fetched at a higher level, and then passed down into `WorkerdApi`.